lms/services/canvas_api/client.py

```diff
@@ -276,15 +276,80 @@ def group_category_groups(self, group_category_id):
             schema=self._ListGroups,
         )
 
-    def course_groups(self, course_id, only_own_groups=True):
+    def course_groups(self, course_id, only_own_groups=True, include_users=False):
+        """
+        Get all the groups of a course.
+
+        :param course_id: Course canvas ID
+        :param only_own_groups: Only return groups the current users belongs to
+        :param include_users: Optionally include all the users in each group
+        """
+        params = {"only_own_groups": only_own_groups}
+        if include_users:
+            params["include[]"] = "users"
+
         return self._client.send(
             "GET",
             f"courses/{course_id}/groups",
-            params={"only_own_groups": only_own_groups},
+            params=params,
             schema=self._ListGroups,
         )
 
+    def current_user_groups(self, course_id, group_category_id=None):
+        """
+        Get all groups the current user belongs in a course and optionally in a group_category.
+
+        :param course_id: Course canvas ID
+        :param group_category_id: Only return groups that belong to this group category
+        """
+        user_groups = self.course_groups(course_id, only_own_groups=True)
+
+        if group_category_id:
+            user_groups = [
+                g for g in user_groups if g["group_category_id"] == group_category_id
+            ]
+
+        return user_groups
+
+    def user_groups(self, course_id, user_id, group_category_id=None):
+        """
+        Get the groups a `user_id` belongs to in an specific `course_id`.
+
+        Optionally return only the groups that belong to a `group_category_id`
+        """
+        canvas_groups = self.course_groups(
+            course_id, only_own_groups=False, include_users=True
+        )
+        groups = []
+        # Look for the student we are grading in all the groups
+        for group in canvas_groups:
+            if group_category_id and group["group_category_id"] != group_category_id:
+                continue
+
+            for user in group["users"]:
+                if user["id"] == user_id:
+                    groups.append(group)
+                    if group_category_id:
+                        # If we are filtering by `group_category_id` there
+                        # can be only one group the user belongs to in the same group set
+                        return groups
```

Comment:
I think we should maybe remove this. The code will be simpler. Even if we think it's correct, do we need to encode this assumption that a user can only belong to one group?

If it turns out (now or in the future) that a user can somehow belong to more than one group then:

* If we just return the list of multiple groups that the user belongs to then the client will just show all of those groups in its menu, which is the same thing that happens when a user belongs to multiple sections
* On the other hand if we've coded in the assumption that a user can only belong to one group then we'll end up only showing the first group even if the user belongs to multiple, which is weird

So even though this probably doesn't make a difference (since users probably can't belong to multiple groups anyway) it's safer not to assume it and might actually make the code simpler.

What do you think?